RubberduckTests/Inspections/DuplicatedAnnotationInspectionTests.cs

```diff
@@ -0,0 +1,138 @@
+using RubberduckTests.Mocks;
+
+namespace RubberduckTests.Inspections
+{
+    [TestFixture]
+    public class DuplicatedAnnotationInspectionTests
+    {
+        [Test]
+        [Category("Inspections")]
+        public void AnnotationDuplicated_ReturnsResult()
+        {
+            const string inputCode = @"
+'@Obsolete
+'@Obsolete
+Public Sub Foo()
+End Sub";
+
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new DuplicatedAnnotationInspection(state);
+                var inspectionResults = inspection.GetInspectionResults(CancellationToken.None);
+
+                Assert.AreEqual(1, inspectionResults.Count());
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void AnnotationDuplicatedTwice_ReturnsSingleResult()
+        {
+            const string inputCode = @"
+'@Obsolete
+'@Obsolete
+'@Obsolete
+Public Sub Foo()
+End Sub";
+
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new DuplicatedAnnotationInspection(state);
+                var inspectionResults = inspection.GetInspectionResults(CancellationToken.None);
+
+                Assert.AreEqual(1, inspectionResults.Count());
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void AnnotationNotDuplicated_DoesNotReturnResult()
+        {
+            const string inputCode = @"
+'@Obsolete
+Public Sub Foo()
+End Sub";
+
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new DuplicatedAnnotationInspection(state);
+                var inspectionResults = inspection.GetInspectionResults(CancellationToken.None);
+
+                Assert.AreEqual(0, inspectionResults.Count());
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void AnnotationAllowingMultipleApplicationsDuplicated_DoesNotReturnResult()
+        {
+            const string inputCode = @"
+'@Ignore(Bar)
+'@Ignore(Baz)
+Public Sub Foo()
+End Sub";
+
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new DuplicatedAnnotationInspection(state);
+                var inspectionResults = inspection.GetInspectionResults(CancellationToken.None);
+
+                Assert.AreEqual(0, inspectionResults.Count());
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void MemberAnnotationDuplicatedForModuleDeclaration_DoesNotReturnResult()
```

Comment:
I don't think we should check that. This intermingles the responsibility of determining legality (member annotations on members, module annotations on modules) with the responsibility of determining duplication.

These are separate things. I'd expect the code tested here to spawn the following inspection results:

 - IllegalAnnotation: member Annotation "TestInitialize" used at module level (2)
 - DuplicateAnnotation: Annotation "TestInitialize" is not intended to be specified mutliple times

That'd simplify the condition mentioned above